### PR TITLE
#2778 Fixed SB Label Providers

### PR DESCRIPTION
**Pull request by etraisnel2**

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java [hunk 1]
```diff
@@ -13,39 +13,30 @@
...
 
   @Override
   public Image getImage(Object element) {
-    if (element instanceof EObjectWrapper) {
-      element = ((BrowserElementWrapper) element).getElement();
-    }
-    return super.getImage(element);
-  }
+    // Do not unwrap category or primitive wrappers, as they have a specific image
+    if (element instanceof CategoryWrapper || element instanceof PrimitiveWrapper) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java:32]: a bit weird to have a empty if.
maybe move the comment below, or a !(element CategoryWapper && !xxx)

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java [hunk 2]
```diff
@@ -13,39 +13,30 @@
...
-      element = ((BrowserElementWrapper) element).getElement();
+      // Unwrap all other wrappers
+      Object wrappedElement = ((EObjectWrapper) element).getElement();
+      return super.getImage(wrappedElement);
     }
-    return super.getText(element);
+    return super.getImage(element);
   }
-
 }
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java:42]: note that you no longer provide getStyledText or getText for EObjectWrapper

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserTree.java [hunk 3]
```diff
@@ -1,150 +1,174 @@
...
+      if (element.getData() != null) {
+        allItems.add(element);
+        getAllItems(element, allItems);
+      }
+    }
+  }
+
+  @Override
+  protected void handleLabelProviderChanged(LabelProviderChangedEvent event) {
+    // TODO Auto-generated method stub
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserTree.java:154]: can you remove todo

**Review by pdulth — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java (hunk 2 above)
Comment (etraisnel2) [reply core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java:42]: Yes, SemanticBrowserDecoratingLabelProvider doesn't provide it anymore, so we let SemanticBrowserLabelProvider handle it 
 getText is used by the getStyledText only
 where they are properly unwrapped

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java (hunk 1 above)
Comment (etraisnel2) [reply core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java]: removed it totally as it was irrelevant

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserLabelProviderFactoryImpl.java [hunk 4]
```diff
@@ -51,7 +50,7 @@ public ILabelProvider getDiagramLabelProvider() {
    * @see org.polarsys.capella.common.ui.toolkit.browser.label.provider.factory.AbstractLabelProviderFactory#getReferencedLabelProvider()
```
Comment (mchoubby) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserLabelProviderFactoryImpl.java:50]: update doc

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserLabelProviderFactoryImpl.java [hunk 5]
```diff
@@ -62,7 +61,7 @@ public ILabelProvider getReferencedLabelProvider() {
    * @see org.polarsys.capella.common.ui.toolkit.browser.label.provider.factory.AbstractLabelProviderFactory#getReferencingLabelProvider()
```
Comment (mchoubby) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserLabelProviderFactoryImpl.java:61]: same

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserTree.java [hunk 6]
```diff
@@ -1,150 +1,173 @@
...
+        getAllItems(element, allItems);
+      }
+    }
+  }
+
+  @Override
+  protected void handleLabelProviderChanged(LabelProviderChangedEvent event) {
+    super.handleLabelProviderChanged(event);
+    Object eventTarget = event.getElement();
+    List<Object> additionalElements = new ArrayList<Object>();
```
Comment (mchoubby) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserTree.java:156]: juste <>

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java [hunk 7]
```diff
@@ -13,39 +13,24 @@
...
 
 import org.eclipse.jface.viewers.ILabelProvider;
-import org.eclipse.jface.viewers.StyledString;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.ui.internal.navigator.NavigatorDecoratingLabelProvider;
-import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.BrowserElementWrapper;
 import org.polarsys.capella.common.ui.toolkit.browser.content.provider.wrapper.EObjectWrapper;
 
-public class SemanticBrowserDecoratingLabelProvider extends  NavigatorDecoratingLabelProvider{
+@SuppressWarnings("restriction")
```
Comment (mchoubby) [inline core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java:20]: la mettre sur la ligne/méthode qui cause le warning sera mieux

**Review by mchoubby — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java (hunk 7 above)
Comment (etraisnel2) [reply core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/label/provider/SemanticBrowserDecoratingLabelProvider.java:20]: C'est NavigatorDecoratingLabelProvider qui cause le warning

**Review by tguiu — APPROVED** [review]

**Pull request merged**